src/Core/Services/StateService.cs

```diff
@@ -1,44 +1,1601 @@
-﻿using Bit.Core.Abstractions;
+﻿using System;
+using Bit.Core.Abstractions;
 using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
 using System.Threading.Tasks;
+using Bit.Core.Enums;
+using Bit.Core.Models.Data;
+using Bit.Core.Models.Domain;
+using Bit.Core.Models.View;
+using Bit.Core.Utilities;
+using Newtonsoft.Json;
 
 namespace Bit.Core.Services
 {
     public class StateService : IStateService
```

Comment:
We already talked about naming improvements but now looking a bit better into it, it's mixing a couple of concerns, could be 2-3 services instead.